### PR TITLE
CI: use pytest-timeout to avoid windows timeouts

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Broken off from #44429, where I haven't seen any of the windows timeouts over quite a few runs.

### CODE REVIEW

File: pandas/tests/io/test_user_agent.py [hunk 1]
```diff
@@ -13,6 +13,10 @@
 import pandas as pd
 import pandas._testing as tm
 
+# Troubleshooting build failures on Windows that tentatively look like
```
Comment (mroeschke) [inline pandas/tests/io/test_user_agent.py:16]: My bet is this file as well.

The `threading.Thread` and `http.server.HTTPServer` can both lock I think.

I am attempting to convert this to a process we can terminate instead.

Comment (jbrockmendel) [reply pandas/tests/io/test_user_agent.py:16]: OK.  if you have a better solution on deck, should we close this?

Comment (mroeschke) [reply pandas/tests/io/test_user_agent.py:16]: Probably best to keep it open to hopefully prove it was this file with a couple of CI passes.

Still trying to evaluate if a process is any better.

**Comment by jreback** [comment]
thanks, good to have in any event

**Pull request merged**